Hunting Queries/Microsoft 365 Defender/Command and Control/SuspiciousNetworkConnection-Googleapis.yaml

```diff
@@ -0,0 +1,45 @@
+id: fe4f4336-a2e7-4cae-af43-3210106475be
+name: Suspicious network connection to Google APIs - Possible GC2-Sheet C2
+description: |
+  This query will look for connections made to several Google cloud service API endpoints which are not associated with a legitimate browser or application. The query was inspired by the GC2-Sheet project: https://github.com/looCiprian/GC2-sheet. GC2-Sheet is a C2 platform designed to leverage Google Sheets and Google Drive SaaS applications.
+requiredDataConnectors:
+  - connectorId: MicrosoftThreatProtection
+    dataTypes:
+    - DeviceNetworkEvents
+tactics:
+  - Command and control
+technique:
+  - T1102
+query: |
+  let excludedProcessFileNames = datatable (browser:string)["teams.exe","GoogleUpdate.exe","outlook.exe","msedge.exe","chrome.exe","iexplorer.exe","brave.exe","firefox.exe"]; //add additional browsers, mail clients, and legitimate google desktop applications where needed for exclusion in your environment
+  DeviceNetworkEvents
+  | where not(InitiatingProcessFileName has_any (excludedProcessFileNames))
+  | where RemoteUrl has_any ("oauth2.googleapis.com","sheets.googleapis.com","drive.googleapis.com") and isnotempty(InitiatingProcessFileName)
+  | summarize visitedURLs=make_list(RemoteUrl,256) by ActionType, DeviceName, InitiatingProcessAccountName, InitiatingProcessAccountUpn, InitiatingProcessAccountDomain, InitiatingProcessParentFileName, InitiatingProcessFileName, InitiatingProcessSHA256, InitiatingProcessSHA1
+  | project ActionType, DeviceName, InitiatingProcessAccountName, InitiatingProcessAccountUpn, InitiatingProcessAccountDomain, InitiatingProcessParentFileName, InitiatingProcessFileName, InitiatingProcessSHA256, InitiatingProcessSHA1, visitedURLs, Connections=array_length(visitedURLs)
+  //| where visitedURLs contains "oauth2.googleapis.com" and visitedURLs has_any ("sheets.googleapis.com","drive.googleapis.com") // may allow for higher fidelity as the GC2 go application communicates to both the google drive folder and sheets API.
+  | extend Account_0_Name = InitiatingProcessAccountUpn
+  | extend Host_0_HostName = DeviceName
+  | extend File_0_Name = InitiatingProcessFileName
+  | extend FileHash_0_Name = InitiatingProcessSHA256
+  | extend FileHash_1_Algorithm = "SHA256"
```

Comment:
FileHash_0_Algorithm is required. The use of 1 is when you have more than one FileHash to map.
Also, you would have to extend a new column with the value "SHA256" and then assign it to FileHash_0_Algorithm. 

Also, FileHash has no identifier as Name. There are only two, Value and Algorithm.

Something like this:
  | extend HashAlgorithm= "SHA256"
  | extend Account_0_Name = InitiatingProcessAccountUpn
  | extend Host_0_HostName = DeviceName
  | extend File_0_Name = InitiatingProcessFileName
  | extend FileHash_0_Value = InitiatingProcessSHA256
  | extend FileHash_0_Algorithm = HashAlgorithm
  
  Since this is present in all the queries, the solution is required to be updated across.